### ISSUE TITLE
サイト名を設定し忘れたままリリースしてしまうサイトがある

### ISSUE THREAD

**Issue by yutorishine** [baserproject/basercms#1790]
## 概要

https://www.google.com/search?q=baserCMS+inc&rlz=1C5CHFA_enJP834JP835&ei=TYzJYcnYB5j8wQOb0JXgBg&start=0&sa=N&ved=2ahUKEwiJs8-I2oP1AhUYfnAKHRtoBWw4MhDy0wN6BAgBEDs&biw=1920&bih=1089&dpr=1

「baserCMS inc」で検索すると、サイト名がデフォルトのまま公開してしまっているサイトが結構あります。
サイト名は、インストール段階で決定しているものも多いと思うので、インストール時に設定させる方がいいのではないでしょうか？

**Comment by yama** [comment]
https://www.vektor-inc.co.jp/post/howto-sitehealth/
WordPressの「サイトヘルス」みたいな機能があるといいかもと思いました。
サイト名以外にもリリース時のチェック項目は多そうなので・・
faviconもbaserCMSのまま使ってるサイトが多いですね、、

**Comment by yutorishine** [comment]
@yama 
サイト基本設定でサイト名変更しない人って、サイトヘルス的なものも使わない気がします。
インストール時に強制的に入力させるのが、やはり一番確実かと。

**Comment by GUSSAN-BEEDAN** [comment]
「Webサイト名」欄および「Webサイトタイトル」欄に引用されているデフォルトは、
ずいぶん以前は、setting.php内に以下のように記述されていましたが、
```
/**
 * アプリケーション基本設定
 */
	$config['BcApp'] = array(
		// デフォルトタイトル設定（インストールの際のエラー時等DB接続まえのエラーで利用）
		'title'				=> 'コーポレートサイトにちょうどいいCMS - baserCMS - ',
```
		
現在は、以下です。

```
/**
 * アプリケーション基本設定
 */
$config['BcApp'] = [
	// デフォルトタイトル設定（インストールの際のエラー時等DB接続まえのエラーで利用）
	'title' => __d('baser', 'baserCMS'),
```
つまり、インストール後のタイミングで、本来入力すべきタイトルを放置し忘れているのではありません。
「baserCMS inc.」は、nada-iconsテーマの初期データ内にサンプルとして記述されているもので、この辺りの扱いを改善しないといたちごっこと思います。

**Comment by yutorishine** [comment]
@GUSSAN-BEEDAN 

なるほど、テーマインストール時に上書きされてしまうわけですね。「baserCMS.inc」はnada-iconなんですね。
BcSampleで試すと「baserCMSサンプル」でした。
しかし、テーマの情報で既に設定されているサイト名を上書きする必要があるのか・・・ってのは個人的には疑問ではあります。

- インストール時にサイト名設定する

- マーケット等に既に出ているテーマは仕方ないとして、梱包されているテーマに関しては、インストール時にサイト名を上書きしないような仕様にする
という対応をセットで行う必要あるんじゃないか・・・と思います。

**Comment by GUSSAN-BEEDAN** [comment]
確かに、テーマを変えるたびにいちいち「Webサイト名」欄および「Webサイトタイトル」欄を上書きされるのは如何なものかとは思いますが、本来ユーザー側で入力すべきこれらの設定を強制的に入れさせるというのは、個人的には必要ないかと思います。

**Comment by yutorishine** [comment]
@GUSSAN-BEEDAN 

そうですね、強制入力欄が増えるのは自分も嫌です。
任意入力としてインストール時のサイト名入力欄を作成、
入力がなかった場合は、おっしゃるように下記の対応でいいかと思います。
Wordpressなんかはそんな感じです。

> setting.phpで入力を促すようなサンプル（たとえば、「サイト名を入力してください。」）としておく程度で良いのではないでしょうか。
> テーマファイルの初期データで上書きされないような配慮さえあれば十分かと。

**Comment by kaburk** [comment]
改善したらよさそうなポイントとして

・Webサイト名のデフォルト値を変更する（サイト名を入力してください など）
・テーマの初期データ読み込みの機能ではサイト名を変更しないようにする

という感じでしょうか。

あとは、サイト基本設定のWebサイト名以外の基本項目（サイト基本キーワード、サイト基本説明文などsite_configsテーブルのデータいくつか）も同様な動作にしたほうが良さそうですね。

**Comment by yutorishine** [comment]
@kaburk 

それがベストな気がします。
wordpress参考にですね。

> サイト基本設定のWebサイト名以外の基本項目（サイト基本キーワード、サイト基本説明文などsite_configsテーブルのデータいくつか）

「これ今入力しなきゃいけないのかな・・・？」みたいな混乱防止のため、「後からでも更新できますよ」的な説明文も添えてあるといいかもしれません。
